### DISCUSSION TITLE
how to add page2 in pdf pycairo?

### DISCUSSION THREAD

**Discussion by maheshs11** [zauberzeug/nicegui#2240] [Q&A] [answered]
### Question

```py
def update() -> None:
    preview.content = generate_svg()
    PDF_PATH.write_bytes(generate_pdf())

    preview1.content = generate_svg()
    PDF_PATH.write_bytes(generate_pdf())


with ui.row():
    with ui.column():
        name = ui.input('Name', placeholder='Enter your name', on_change=update)
        email = ui.input('E-Mail', placeholder='Enter your E-Mail address', on_change=update)
    preview = ui.html().classes('border-2 border-gray-500')
    preview1 = ui.html().classes('border-2 border-gray-500')

    update()
    ui.button('Download PDF', on_click=lambda: ui.download(PDF_PATH)).bind_visibility_from(name, 'value')

ui.run()
```

```py
ui.button('Download PDF', on_click=lambda: (ui.download(PDF_PATH1), ui.download(PDF_PATH2)).bind_visibility_from(name, 'value'))
```

i can download separately like this, is there any way to do in same file?

---

Remark from the maintainer: @maheshs11 is referring to the ["Generate PDF" example](https://github.com/zauberzeug/nicegui/blob/main/examples/generate_pdf/).

**Comment by rodja** [comment]
Sorry, I do not understand your question. You want to download two files as one? You mean as zip file?

**Reply by maheshs11** [reply]
no, i want two A4 sheet size in a single pdf.

i guess i have to use external pdf merge, right?
![image](https://github.com/zauberzeug/nicegui/assets/7268915/0f1c5c50-60c8-4f16-97ba-db5d0451d41f)

**Comment by falkoschindler** [comment, marked as answer]
You can write multiple pages to a PDF using `surface.show_page()`. Here I'm simply duplicating the first page:

```py
def generate_pdf() -> bytes:
    output = BytesIO()
    surface = cairo.PDFSurface(output, 300, 200)
    draw(surface)
    surface.show_page()
    draw(surface)
    surface.finish()
    return output.getvalue()
```

Please refer to the [Pycairo documentation](https://pycairo.readthedocs.io/) for more information.

**Reply by maheshs11** [reply]
thanks!
